daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/MapKOps.scala

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2019 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.daml.lf.data
+
+import scala.language.higherKinds
+import scala.collection.GenTraversableOnce
+import scala.collection.immutable.{Map, MapLike}
+
+/** Halfway between the *-kinded MapLike and *->*->*-kinded MapOps. */
+trait MapKOps[K, +V, +This[+TV] <: Map[K, TV] with MapKOps[K, TV, This]] extends MapLike[K, V, This[V]] { this: This[V] =>
```

Comment:
@remyhaemmerle-da Unfortunately this was left out of the Scala 2.[8,13) collections library; I leave it here as the memory of a dream of a better world. (And also any custom map we define should use it.)

Note how trivial it would be to make `+` appear to return `TextMap` were `MapKOps` incorporated into the standard library.